### ISSUE TITLE
Feature request: range configurability for light sources (notably torches)

### ISSUE THREAD

**Issue by nbusseneau** [shudnal/GammaOfNightLights#4]
Hey, love the mod, thanks a lot!

I've settled on using night settings with 0.65 luminance / 1.75 moonlight to get a more immersive survival experience when moonlight is blocked by trees in the forest while retaining a chill experience in open environments, with the idea being that if you want to go out at night in the forest you MUST prepare light sources. However, vanilla torches are basically useless as portable light sources.

The [Torch Light Mod](https://github.com/aedenthorn/ValheimMods/blob/master/TorchMod/TorchMod.cs) allows to counteract this, however as you may know Aedenthorn does not upload mods to Thunderstore, which makes it annoying to integrate in modpacks.

What do you think about adding configurability for range of light sources (and in particular, torches) as part of the mod? I personally think it would make sense to add a setting `Light source range multiplier` to allow for scaling range on all light sources while retaining vanilla ratios, where some have higher distances than others.

**Comment by shudnal** [comment]
It looks like something a bit outside of the mod's purpose. Configuring only torches range would be somewhat a half-measures. And configuring the other light values will bloat the configs.

If your only goal is to add mod to modpack I think the solution is to reupload a mod from nexus as is. Like it was already done before https://thunderstore.io/c/valheim/p/dyne/TorchLightMod/ but without updates sadly. Nothing stops you from reuploading the mod with exact the same name.

I've also made https://thunderstore.io/c/valheim/p/shudnal/CircletExtended/ to make night traversal more fun.

But you made me thinking of making another torches/lights mod. I had a couple of ideas but it will not be any soon.

**Comment by nbusseneau** [comment]
> And configuring the other light values will bloat the configs.

Aye, I agree, that's why I was thinking more along the lines of a single setting that scales all light sources at once.

> But you made me thinking of making another torches/lights mod. I had a couple of ideas but it will not be any soon.

Care to elaborate on this? 🤔

**Comment by shudnal** [comment]
https://github.com/shudnal/GammaOfNightLights/assets/20457922/354e9b12-7ecb-482f-a8b0-6b0885dcaa11

Desired effect is like this. Basically a wisplight but with warm temperature, reduced intensity, increased range, less particle effects. Like a firefly. Allowing night traversal with weapons in hand and not to stumble across obstacles in close range but without a visibility on distance. Something better than torch but less powerful than extended circlet.

But I didn't come up yet how to integrate that in gameplay properly. Should it be an wearable item with constant effect or consumable with status effect like mead or something like "build a torch then break a torch to release a light for 5 min".

**Comment by nbusseneau** [comment]
Ah, I see. I think I personally prefer the immersion of torches, so no opinion 😄
